docs/codeql/codeql-language-guides/analyzing-data-flow-in-python.rst

```diff
@@ -0,0 +1,356 @@
+
+These predicates are defined in the configuration:
+
+-  ``isSource`` - defines where data may flow from.
+-  ``isSink`` - defines where data may flow to.
+-  ``isBarrier`` - optionally, restricts the data flow.
+-  ``isAdditionalFlowStep`` - optionally, adds additional flow steps.
+
+The characteristic predicate (``MyDataFlowConfiguration()``) defines the name of the configuration, so ``"..."`` must be replaced with a unique name (for instance the class name).
+
+The data flow analysis is performed using the predicate ``hasFlow(DataFlow::Node source, DataFlow::Node sink)``:
+
+.. code-block:: ql
+
+   from MyDataFlowConfiguation dataflow, DataFlow::Node source, DataFlow::Node sink
+   where dataflow.hasFlow(source, sink)
+   select source, "Dataflow to $@.", sink, sink.toString()
+
+Using global taint tracking
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Global taint tracking is to global data flow what local taint tracking is to local data flow. That is, global taint tracking extends global data flow with additional non-value-preserving steps. The global taint tracking library is used by extending the class ``TaintTracking::Configuration``:
+
+.. code-block:: ql
+
+   import python
+
+   class MyTaintTrackingConfiguration extends TaintTracking::Configuration {
+     MyTaintTrackingConfiguration() { this = "..." }
+
+     override predicate isSource(DataFlow::Node source) {
+       ...
+     }
+
+     override predicate isSink(DataFlow::Node sink) {
+       ...
+     }
+   }
+
+These predicates are defined in the configuration:
+
+-  ``isSource`` - defines where taint may flow from.
+-  ``isSink`` - defines where taint may flow to.
+-  ``isSanitizer`` - optionally, restricts the taint flow.
+-  ``isAdditionalTaintStep`` - optionally, adds additional taint steps.
+
+Similar to global data flow, the characteristic predicate (``MyTaintTrackingConfiguration()``) defines the unique name of the configuration and the taint analysis is performed using the predicate ``hasFlow(DataFlow::Node source, DataFlow::Node sink)``.
+
+Flow sources
+~~~~~~~~~~~~
+
+The data flow library contains some predefined flow sources. The class ``RemoteFlowSource`` (defined in module ``semmle.python.dataflow.new.RemoteFlowSources``) represents data flow from remote network inputs. This is useful for finding security problems in networked services.
+
+For global flow, it is also useful to restrict sources to instances of ``LocalSourceNode``. The predefined sources generally do that.
+
+Example
+~~~~~~~
+
+This query shows a data flow configuration that uses all network input as data sources:
+
+.. code-block:: ql
+
+   import python
+   import semmle.python.dataflow.new.RemoteFlowSources
+
+   class MyDataFlowConfiguration extends DataFlow::Configuration {
+     MyDataFlowConfiguration() {
+       this = "..."
+     }
+
+     override predicate isSource(DataFlow::Node source) {
+       source instanceof RemoteFlowSource
+     }
+
+     ...
+   }
+
+Class hierarchy
+~~~~~~~~~~~~~~~
```

Comment:
Should this section maybe come before the two examples? (that would feel more natural to me)

Oh, I see that C# does the same as we do right now... feel free to just leave it like this :+1:

Reply:
I actually already changed this :-)